src/linker/Linker.Dataflow/MethodBodyScanner.cs

```diff
@@ -531,8 +531,9 @@ public void Scan (MethodBody methodBody)
 						// Pop function pointer
 						PopUnknown (currentStack, 1, methodBody, operation.Offset);
 
-						// Push return value
-						if (signature.ReturnType.MetadataType != MetadataType.Void)
+						// Push return value - BUG: signature.ReturnType.MetadataType sometime doesn't show void for valid cases
+						// Revisit after https://github.com/mono/linker/issues/2090
+						if (signature.ReturnType.GetElementType ().MetadataType != MetadataType.Void)
```

Comment:
We don't want to treat `void*` as `void` here. AFAIK GetElementType will strip off the pointerness. This should `Resolve` the return type or whatever is the convenient thing to do for it in Cecil.

Reply:
`Resolve` will also strip off the pointerness (it does `GetElementType`). I think we need to check the `GetElementType` only if `returnType.IsOptionalModifier`.